tests/bulkwritecommand/bulkwritecommand-deleteOne_error-004.phpt

```diff
@@ -0,0 +1,33 @@
+--TEST--
+MongoDB\Driver\BulkWriteCommand::deleteOne() prohibits PackedArray for document values
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$bulk = new MongoDB\Driver\BulkWriteCommand;
+
+echo throws(function() use ($bulk) {
+    $bulk->deleteOne(NS, MongoDB\BSON\PackedArray::fromPHP([]));
+}, MongoDB\Driver\Exception\UnexpectedValueException::class), "\n";
+
+echo throws(function() use ($bulk) {
+    $bulk->deleteOne(NS, [], ['collation' => MongoDB\BSON\PackedArray::fromPHP([])]);
+}, MongoDB\Driver\Exception\UnexpectedValueException::class), "\n";
+
+// @TODO: ALMOST: Got MongoDB\Driver\Exception\InvalidArgumentException - expected MongoDB\Driver\Exception\UnexpectedValueException
+// Expected "hint" option to yield string or document but got "array"
+echo throws(function() use ($bulk) {
+    $bulk->deleteOne(NS, [], ['hint' => MongoDB\BSON\PackedArray::fromPHP([])]);
+}, MongoDB\Driver\Exception\UnexpectedValueException::class), "\n";
```

Comment:
The exception type is different:
- Collection BulkWrite: `UnexpectedValueException`
- Client BulkWriteCommand: `InvalidArgumentException`

Should we use the same.

Reply:
Technically, the `InvalidArgumentException` is the correct one, but we shouldn't change the legacy bulk write classes to throw different exceptions. I think we can adapt the test and leave the exception classes as-is.